src/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Write.HandleDictionary.cs

```diff
@@ -79,7 +79,7 @@ public static partial class JsonSerializer
                     var enumerator = (IDictionaryEnumerator)state.Current.CollectionEnumerator;
                     object value = enumerator.Value;
                     state.Push(elementClassInfo, value);
-                    state.Current.KeyName = (string)enumerator.Key;
+                    state.Current.KeyName = options.DictionaryKeyPolicy?.ConvertName((string)enumerator.Key) ?? (string)enumerator.Key;
```

Comment:
Could you please add the proper condition to not apply this when writing a Dictionary annotated with `JsonExtensionData`? 
In addition it would be nice to have a test that verifies above statement by checking that keys in `JsonExtensionData` are not affected by `DictionaryKeyPolicy`.

Reply:
@Jozkee , thank for reviewing. Could please clarify, I need add a condition for checking ExtensionDataStatus? Like was done here: https://github.com/dotnet/corefx/blob/master/src/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Write.HandleDictionary.cs#L152
Or I need check dictionary for existence JsonExtensionDataAttribute and ignore DictionaryKeyPolicy for all keys in case when attribute is presented?
Suppose we have a class with property like this:
<pre>
private class ClassWithExtensionDataProperty
 {
  [JsonExtensionData]
  public Dictionary Data { get; set; }
 }
</pre>
if we create an obj and initialize dictionary with KeyPolicy:
<pre>
  var obj = new ClassWithExtensionDataProperty();
  obj.Data = new Dictionary();
  obj.Data["KeyDict"] = new Dictionary()
   {
      { "KeyString", "text" },
      { "KeyNumber", 1000 },
      { "KeyBool", true }
 };
  string json = JsonSerializer.Serialize(obj, new JsonSerializerOptions()
  {
     DictionaryKeyPolicy = JsonNamingPolicy.CamelCase,
     WriteIndented = true 
});
</pre>
The resulted json will be :
<pre>
{
 "KeyDict": {
    "keyString": "text",
    "keyNumber": 1000,
    "keyBool": true
    }
}
</pre>
Because DictionaryKeyPolicy affects all dictionary keys ( I 've got this result after reverting my changes).

Reply:
> I need add a condition for checking ExtensionDataStatus? Like was done here

Yes.

In your sample, try adding KeyString or KeyInt (or any other primitive) to your `Data` dictionary and you will see that their name will no be affected by DictionaryKeyPolicy. I think it is fine that keys of a dictionary that is part the ExtensionData are affected, but keys directly on the ExtensionData must remain unchanged.

Now, with your changes, you are missing said discrimination which will be fixed by adding a condition similar to [here](https://github.com/dotnet/corefx/blob/master/src/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Write.HandleDictionary.cs#L152) (as you mention).

The reasoning for this is that if you alter a key that is part of ExtensionData, then your payload will no be able to round-trip, see https://github.com/dotnet/corefx/issues/41780.

Reply:
@Jozkee, condition added.
Thank you for reviewing.